### DISCUSSION TITLE
Labels are not properly applied

### DISCUSSION THREAD

**Discussion by munjalpatel** [fission/fission#2141] [Q&A]
**Fission/Kubernetes version**

<pre>
$ fission --version

client:
  fission/core:
    BuildDate: "2021-08-02T11:44:22Z"
    GitCommit: f4702fe066c48e05cd5c093311fe95e4a6a971a4
    Version: 1.14.0
server:
  fission/core:
    BuildDate: "2021-08-02T09:30:56Z"
    GitCommit: f4702fe066c48e05cd5c093311fe95e4a6a971a4
    Version: 1.14.0

$ kubectl version

Client Version: version.Info{Major:"1", Minor:"20", GitVersion:"v1.20.0", GitCommit:"af46c47ce925f4c4ad5cc8d1fca46c7b77d13b38", GitTreeState:"clean", BuildDate:"2020-12-08T17:59:43Z", GoVersion:"go1.15.5", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"21+", GitVersion:"v1.21.2-13+d2965f0db10712", GitCommit:"d2965f0db1071203c6f5bc662c2827c71fc8b20d", GitTreeState:"clean", BuildDate:"2021-06-26T00:59:14Z", GoVersion:"go1.16.5", Compiler:"gc", Platform:"linux/amd64"}
</pre>

**Kubernetes platform**
AWS EKS

**Describe the bug**
1. If no label is applied to `env` and a label is applied to `fn`, the label is not available in the function pod.
2. If lavel is applied to `env` and a label is also applied to `fn`, the label in the function pod has a value from `env` and not `fn`.

**To Reproduce**
For 1:
1. Create an env and do not add any label
2. Create a function with `--labels "app=my-app"`

Notice that the function pod does not have the label `app: my-app`

For 2:
1. Create an env with `--labels "app=go-env"`
2. Create a function with `--labels "app=my-app"`

Notice that the function pod has the label `app: go-env`

**Expected result**
1. If no label is applied to `env` but a label has been added to a `fn`, the function pod should have that label applied.
2. If both `env` and `fn` have the same label, the label value on the function pod should take precedence.

cc: @vishal-biyani @sanketsudake

**Comment by sanketsudake** [comment]
Hey @munjalpatel,

As of now in the 1.14 release, we are propagating environment variables to the function pod and function labels.
This is primarily because pods are specialized in the context of the request. We are not sure unless the function is invoked which pod would get specialized in what context.

Adding function labels/annotations to the function pod is possible, though many controllers won't be able to react in the context of function requests, so action-based labels/annotations like injecting secret/sidecar won't happen in real-time.

We would revisit this later, but currently, we are propagating env labels to the pods in the case of newdeploy and pool manager.
